gitlab/resource_gitlab_topic.go

```diff
@@ -0,0 +1,124 @@
+package gitlab
+
+import (
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabTopic() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabTopicCreate,
+		Read:   resourceGitlabTopicRead,
+		Update: resourceGitlabTopicUpdate,
+		Delete: resourceGitlabTopicDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceGitlabTopicCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	options := &gitlab.CreateTopicOptions{
+		Name: gitlab.String(d.Get("name").(string)),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		options.Description = gitlab.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] create gitlab topic %s", *options.Name)
+
+	topic, _, err := client.Topics.CreateTopic(options)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(strconv.FormatInt(int64(topic.ID), 10))
+
+	return resourceGitlabTopicRead(d, meta)
+}
+
+func resourceGitlabTopicRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+
+	topicID, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] read gitlab topic %d", topicID)
+
+	topic, resp, err := client.Topics.GetTopic(topicID, nil)
+	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
```

Comment:
You could make use of `util.is404()` here.

Reply:
From which util package do you mean?

Reply:
Sorry, from this providers package, from `util.go` :)